frontend/src/components/user/edit_profile/EditUserProfile.tsx

```diff
@@ -0,0 +1,358 @@
+import React, { useContext } from "react";
+import { UserContext } from "../../../services/authService";
+import { api, useBetterQuery } from "../../../services/apiService";
+import { Profile, ProfileQuestion } from "../../../models/profile";
+import { Loading } from "../../utils/Loading";
+import { Error } from "../../utils/Error";
+import Container from "react-bootstrap/Container";
+import { PageTitle } from "../../utils/PageTitle";
+import { Form, Formik } from "formik";
+import Card from "react-bootstrap/Card";
+import { Form as ReactBootstrapForm } from "react-bootstrap";
+import { TextFormGroup } from "../../utils/forms/TextFormGroup";
+import Button from "react-bootstrap/Button";
+import { SelectGroup } from "../../utils/forms/SelectGroup";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import { queryCache, useMutation } from "react-query";
+import { ToastContext, ToastLevel } from "../../utils/Toast";
+import { AxiosError } from "axios";
+import * as Yup from "yup";
+import { SelectUsers } from "../../utils/forms/SelectUsers";
+
+// The items in the select.
+const currentAcademicYearItems = new Map([
+    ["1A", <span className="selectgroup-button">1A</span>],
+    ["2A", <span className="selectgroup-button">2A</span>],
+    ["GAP YEAR", <span className="selectgroup-button">Césure</span>],
+    ["3A", <span className="selectgroup-button">3A</span>],
+    ["GRADUATE", <span className="selectgroup-button">Alumni</span>],
```

Comment:
Maybe the span part is a bit redundant

Reply:
Yes, it's an issue with the component which use them… Let's keep it this way, I opened #179